optaplanner-core/src/main/java/org/optaplanner/core/impl/score/stream/bavet/uni/BavetAbstractUniConstraintStream.java

```diff
@@ -144,133 +144,147 @@ public void addChildStream(BavetAbstractUniConstraintStream<Solution_, A> childS
     }
 
     // ************************************************************************
-    // Penalize/reward
+    // Penalize/impactScore
```

Comment:
Penalize without reward is like ying without yang. So let's not remove "reward"

Users should only use penalize and reward, never impactScore.
Both penalize and reward only accept positive matchWeights.

Reply:
Yeah, this is a mistake. It's a side-effect of mass-rename, shouldn't have happened.